pyiron_base/jobs/master/serial.py

```diff
@@ -25,119 +24,37 @@
-        .. attribute:: job_info_str
-
-            short string to describe the job by it is job_name and job ID - mainly used for logging
-
-        .. attribute:: working_directory
-
-            working directory of the job is executed in - outside the HDF5 file
-
-        .. attribute:: path
-
-            path to the job as a combination of absolute file system path and path within the HDF5 file.
-
-        .. attribute:: version
-
-            Version of the hamiltonian, which is also the version of the executable unless a custom executable is used.
-
-        .. attribute:: executable
-
-            Executable used to run the job - usually the path to an external executable.
-
-        .. attribute:: library_activated
-
-            For job types which offer a Python library pyiron can use the python library instead of an external
-            executable.
-
-        .. attribute:: server
-
-            Server object to handle the execution environment for the job.
-
-        .. attribute:: queue_id
-
-            the ID returned from the queuing system - it is most likely not the same as the job ID.
-
-        .. attribute:: logger
-
-            logger object to monitor the external execution and internal pyiron warnings.
-
-        .. attribute:: restart_file_list
-
-            list of files which are used to restart the calculation from these files.
-
-        .. attribute:: job_type
-
-            Job type object with all the available job types: ['ExampleJob', 'SerialMaster', 'ParallelMaster',
-                                                               'ScriptJob', 'ListMaster']
-
-        .. attribute:: child_names
-
-            Dictionary matching the child ID to the child job name.
+    (
+        """
+    The serial master class is a metajob consisting of a dynamic list of jobs
+    which are executed in serial mode.
 
-        .. attribute:: start_job
+    Example:
 
-            The first job of the series.
+    >>> def convergence_goal(self, **qwargs):
+    >>>     if len(self[-1].output.energy_pot) > qwargs['max_steps']:
+    >>>         return None
+    >>>     else:
+    >>>         return self.create_next()
 
-        .. attribute:: input
+    >>> from pyiron import Project
+    >>> pr = Project('TEST')
+    >>> job = pr.create_job("SomeJobClass", "child_test")
+    >>> job_ser = pr.create_job("SerialMaster", "serial_master_test")
+    >>> job_ser.append(job)
+    >>> job_ser.set_goal(convergence_goal, max_steps=10)
+    >>> job_ser.run()
 
-            The input of the start job - the first job of the series.
     """
+        + "\n    Args:"
+        + GenericMaster.__doc__.split("\n    Args:")[-1]
+    )
 
     def __init__(self, project, job_name):
-        self._input = GenericParameters("parameters")  # e.g. convergence goal
 
         super(SerialMasterBase, self).__init__(project, job_name=job_name)
         self.__version__ = "0.3"
 
-        self._output = GenericOutput()
-        self._max_iterations = 100
```

Comment:
Where do you define the maximum number of iterations?

Reply:
By briefly looking over the code I also do not find a use case, so maybe it is good to delete it.